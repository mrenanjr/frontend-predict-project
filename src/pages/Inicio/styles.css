.container-principal {
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.container-principal .card {
    width: 320px;
    margin: 10px;
    padding: 0px;
    background: linear-gradient(180deg, #77787b 0%, rgba(255, 255, 255, 0) 100%), linear-gradient(180deg, rgba(53, 68, 79, 0) 0%, #77787b 100%);
    cursor: pointer;

    transition: 0.2s;
}

.container-principal .card:hover {
    opacity: 0.8;
}

.container-principal .col-md {
    text-align: center;
    display: flex;
    align-items: center;
}

.container-principal .col-md img {
    transition: 0.2s;
    cursor: pointer;
}

.container-principal .col-md img:hover {
    opacity: 0.5;
}

.container-principal .card p {
    text-align: center;

    margin-bottom: 0.5rem;
    text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.container-principal .title {
    font-weight: bold;
    color: white;
}

.container-principal .students {
    font-weight: normal;
    font-size: 18px;
    color: #0F084B;
}

.container-principal .number {
    font-weight: bold;
}