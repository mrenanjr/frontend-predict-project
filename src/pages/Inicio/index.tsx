import React, { useEffect, useState } from 'react';
import Chart from "react-google-charts";
import { Link } from 'react-router-dom';
import api from '../../services/api';
import { Dimmer, Loader } from 'semantic-ui-react'

import './styles.css'

import Footer from '../../components/Footer';
import Header from '../../components/Header';

import Next from '../../assets/arrow_forward-24px.svg';
import Back from '../../assets/arrow_back-24px.svg';

interface Evasao {
    curso_percent: CursoPercent[],
    percent_evasao_total: number,
    quant_aluno_total: number,
    quant_evasao_total: number
};

export interface CursoPercent {
    curso: string,
    curso_abreviado: string,
    percent_evasao: number,
    quant_aluno: number,
    quant_evasao: number,
};

const Inicio = () => {
    const [evasao, setEvasao] = useState<Evasao>({
      curso_percent: [],
      percent_evasao_total: 0,
      quant_aluno_total: 0,
      quant_evasao_total: 0  
    });
    const [loader, setLoader] = useState('');

    useEffect(() => {
        setLoader('active');
        api.get(`evasao?token=${localStorage.getItem('token')}`).then(resp => {
            let result = resp.data;
            let curso_abreviado = ['cc', 'es', 'si'];
            for(let i = 0; i < result.curso_percent.length; i++) {
                result.curso_percent[i].curso_abreviado = curso_abreviado[i];
            }
            setEvasao(result);
        }).catch(err => {
            alert(`Falha ao tentar buscar as informações de evasão dos alunos na api. Error: ${err}`);
        }).finally(() => {
            setLoader('disabled');
        })
    }, []);

    return (
        <>
            <Header />
            <Dimmer className={loader}>
                <Loader size="large" content='Carregando' />
            </Dimmer>
            <main className="container-principal">
                <div className="row">
                    <div className="col-md-12">
                        <p className="title">Sistema de Análise Preditiva</p>
                        <p className="students" style={{ textAlign: 'center' }}>Total de alunos: {evasao.quant_aluno_total}</p>
                    </div>
                    <div className="col-md-12">
                        <div className="row justify-content-sm-center">
                            <div className="col-md" style={{ justifyContent: 'center' }}>
                                <img src={Back} alt="Back"/>
                            </div>
                            {evasao.curso_percent.map((cp, i) => (
                                <Link className="col-md-3 card" key={i} to={{
                                    pathname: `/detalhes/${cp.curso_abreviado}`,
                                    state: cp
                                }}>
                                    <p className="number">{i + 1}º</p>
                                    <p className="course">{cp.curso}</p>
                                    <p className="students">Predição de evasão: {cp.quant_evasao}</p>
                                    <Chart
                                        width={"100%"}
                                        height={"250px"}
                                        chartType="PieChart"
                                        loader={<div style={{ color: 'white' }}>Carregando</div>}
                                        data={[
                                            ['Constância', 'Porcentagem'],
                                            ['Evasão', cp.percent_evasao],
                                            ['Permanência', 100 - cp.percent_evasao],
                                        ]}
                                        options={{
                                            backgroundColor: 'transparent',
                                            pieHole: 0.4,
                                            colors: ['#0067AC', '#BCBCBC'],
                                            legend: {
                                                position: 'bottom',
                                                alignment: 'start',
                                                textStyle: {
                                                    color: 'white'
                                                }
                                            }
                                        }}
                                        rootProps={{ 'data-testid': '1' }}
                                    />
                                </Link>
                            ))}
                            <div className="col-md" style={{ justifyContent: 'center' }}>
                                <img src={Next} alt="Next"/>
                            </div>
                        </div>
                    </div>
                </div>
            </main>
            <Footer />
        </>
    );
}

export default Inicio;