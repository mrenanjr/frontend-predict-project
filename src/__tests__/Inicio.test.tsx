import * as React from 'react';
import { shallow } from 'enzyme';
import Inicio from '../pages/Inicio';

test('Inicio loads corretly', () => {
  const wrapper = shallow(<Inicio />);

  expect(wrapper).toMatchSnapshot();
});