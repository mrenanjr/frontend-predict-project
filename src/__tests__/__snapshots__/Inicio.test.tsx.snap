// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inicio loads corretly 1`] = `
<Fragment>
  <Header />
  <Dimmer
    className=""
  >
    <Loader
      content="Carregando"
      size="large"
    />
  </Dimmer>
  <main
    className="container-principal"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <p
          className="title"
        >
          Sistema de An√°lise Preditiva
        </p>
        <p
          className="students"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Total de alunos: 
          0
        </p>
      </div>
      <div
        className="col-md-12"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-md"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <img
              alt="Back"
              src="test-file-stub"
            />
          </div>
          <div
            className="col-md"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <img
              alt="Next"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
  <Footer />
</Fragment>
`;
